packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/PBSProgressListenerTests.kt

```diff
@@ -167,46 +152,44 @@ class ProgressListenerTests {
             withTimeout(TIMEOUT) {
                 flow.takeWhile { completed -> completed < 3 }
                     .collect { completed ->
-                        realm.writeSampleData(TEST_SIZE, idOffset = (completed + 1) * TEST_SIZE)
+                        realm.writeSampleData(TEST_SIZE)
                         realm.syncSession.uploadAllLocalChangesOrFail()
                     }
             }
         }
     }
 
     @Test
-    @Ignore // https://github.com/realm/realm-core/issues/7627
     fun worksAfterExceptions() = runBlocking {
         Realm.open(createSyncConfig(app.createUserAndLogIn())).use { realm ->
             realm.writeSampleData(TEST_SIZE, timeout = TIMEOUT)
         }
 
         Realm.open(createSyncConfig(app.createUserAndLogin())).use { realm ->
+            val flow = realm.syncSession.progressAsFlow(Direction.UPLOAD, ProgressMode.INDEFINITELY)
             assertFailsWith<RuntimeException> {
-                realm.syncSession.progressAsFlow(Direction.DOWNLOAD, ProgressMode.INDEFINITELY)
-                    .collect {
-                        @Suppress("TooGenericExceptionThrown")
-                        throw RuntimeException("Crashing progress flow")
-                    }
+                flow.collect {
+                    @Suppress("TooGenericExceptionThrown")
+                    throw RuntimeException("Crashing progress flow")
+                }
             }
 
-            val flow = realm.syncSession.progressAsFlow(Direction.DOWNLOAD, ProgressMode.INDEFINITELY)
             withTimeout(TIMEOUT) {
                 flow.first { it.isTransferComplete }
             }
         }
     }
 
     @Test
-    @Ignore // https://github.com/realm/realm-core/issues/7627
     fun worksAfterCancel() = runBlocking {
         Realm.open(createSyncConfig(app.createUserAndLogIn())).use { realm ->
             realm.writeSampleData(TEST_SIZE, timeout = TIMEOUT)
         }
 
         Realm.open(createSyncConfig(app.createUserAndLogin())).use { realm ->
             // Setup a flow that we are just going to cancel
-            val flow = realm.syncSession.progressAsFlow(Direction.DOWNLOAD, ProgressMode.INDEFINITELY)
+            val flow =
+                realm.syncSession.progressAsFlow(Direction.UPLOAD, ProgressMode.INDEFINITELY)
```

Comment:
Shouldn't this be `Direction.DOWNLOAD` I guess it's irrelevant since it will be cancelled ?

Reply:
Due to https://github.com/realm/realm-core/issues/7627 there will not be any initial event for `DOWNLOAD`s, and since the actual downloads/uploads are not relevant for this test, I just updated it to be able to enable the test. 

https://github.com/realm/realm-core/issues/7627 should be merged in Core, so if we bump to the latest version we could enable all the test relying on the initial event again.

Reply:
Bumped https://github.com/realm/realm-kotlin/pull/1746 to include the fix.